Challenge-2/src/helper.ts

```diff
@@ -0,0 +1,67 @@
+import { ethers } from "forta-agent";
+import { UNISWAP_PAIR_ABI } from "./constants";
+import { LRUCache } from "lru-cache";
+
+export default class Helper {
+  private cache: LRUCache<string, [boolean, string, string, string]>;
+  private externalCallCount: number;
+
+  // Constructor to initialize with an ethers provider
+  constructor() {
+    this.cache = new LRUCache<string, [boolean, string, string, string]>({ max: 1000 });
+    this.externalCallCount = 0;
+  }
+
+  // Computes the CREATE2 address for a Uniswap pair
+
+  public getUniswapPairCreate2Address(
+    factoryAddress: string,
+    token0: string,
+    token1: string,
+    fee: number,
+    initcode: string
+  ): string {
+    return ethers.utils.getCreate2Address(
+      factoryAddress,
+      ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(["address", "address", "uint24"], [token0, token1, fee])
+      ),
+      initcode
+    );
+  }
+
+  public getExternalCallCount(): number {
+    return this.externalCallCount;
+  }
+
+  public resetExternalCount() {
+    this.externalCallCount = 0;
+  }
```

Comment:
I think adding class methods that are used only in testing can be avoided.

You can see here for a similar implementation of the class test that uses the cached values
https://github.com/NethermindEth/Forta-Agents/blob/main/PancakeSwap-Bots/Large-Swap/src/data.fetcher.spec.ts#L79-L89